pkg/util/coverage/coverage.go

```diff
@@ -0,0 +1,91 @@
+// +build coverage
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coverage
+
+import (
+	"flag"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"os"
+	"testing"
+	"time"
+)
+
+const flushInterval = 5 * time.Second
+
+var periodChannel chan struct{}
+var coverageFile string
+
+// tempCoveragePath returns a temporary file to write coverage information to.
+// The file is in the same directory as the destination, ensuring os.Rename will work.
+func tempCoveragePath() string {
+	return coverageFile + ".tmp"
+}
+
+// InitCoverage is called from the dummy unit test to prepare Go's coverage framework.
+// Clients should never need to call it.
+func InitCoverage(name string) {
+	// We read the coverage destination in from the KUBE_COVERAGE_FILE env var,
+	// or if it's empty we just use a default in /tmp
+	coverageFile = os.Getenv("KUBE_COVERAGE_FILE")
```

Comment:
Glad to see the runtime configuration options within util.
I wonder if something like this could be use to augment client-go rest api-calls?

Reply:
Use of an env var for config makes me wince a bit.  We _do_ have prior art that uses it, but I'm curious if there's a way we could do this via a command line flag instead. ie: have the dummy test add a command line flag with a default value, call this with the parsed value?

Reply:
adding package scoped flags is generally pretty awful, env are not great for lots of config but it's just one simple string and only when coverage is enabled. doing it this way has the upside of not mutating any global state within the binary. this is probably the way to go imho.

Reply:
Another downside to using a flag instead of an environment variable is that, because this feature is not always compiled in, you now need to be particularly careful about when you are passing the flag. By contrast, the environment variable will be silently ignored when it isn't relevant.

Reply:
OK I'm sold